crates/shared/src/price_estimation.rs

```diff
@@ -152,31 +152,39 @@ impl NativePriceEstimators {
         Self(Vec::new())
     }
 
-    pub fn as_slice(&self) -> &[NativePriceEstimator] {
+    pub fn as_slice(&self) -> &[Vec<NativePriceEstimator>] {
         &self.0
     }
 }
 
 impl Default for NativePriceEstimators {
     fn default() -> Self {
-        Self(vec![NativePriceEstimator::GenericPriceEstimator(
+        Self(vec![vec![NativePriceEstimator::GenericPriceEstimator(
             PriceEstimatorKind::Baseline,
-        )])
+        )]])
     }
 }
 
 impl Display for NativePriceEstimators {
     fn fmt(&self, f: &mut Formatter) -> fmt::Result {
-        let mut it = self.as_slice().iter();
-        if let Some(kind) = it.next() {
-            write!(f, "{kind:?}")?;
-            for kind in it {
-                write!(f, ",{kind:?}")?;
+        if self.0.is_empty() {
+            return f.write_str("None");
+        }
+
+        let mut stage_it = self.as_slice().iter().peekable();
```

Comment:
uber naming nit: I'm not a fan of using `_{type}` as variable names. I think `let mut stages = self.as_slice().iter().peekable()` is more "natural" from a language perspective. Just my opinion though.

Also, I think you may be able to use [`Itertools::join`](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.join) to join strings with separators and might make the code a little nicer. If you want to get really fancy - you can even use [`Itertools::format`](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.format) and [`Itertools::format_with`](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.format_with) to do it without any allocations 😄.

<details><summary>Something like:</summary>

⚠️ Untested code!

```rust
        let formatter = self
            .as_slice()
            .iter()
            .map(|stage| {
                stage
                    .iter()
                    .format_with(",", |estimator, f| f(&format_args!("{:?}", estimator)))
            })
            .format(";");
        write!(f, "{}", formatter)
```

</details>

Note that these are just notes, feel free to merge this PR without